blog/models.py

```diff
@@ -124,9 +130,21 @@ class BlogArticlePage(MixinSeoFields, Page, MixinPageMethods, GoogleAdsMixin):
         FieldPanel("is_main_article"),
         ImageChooserPanel("cover_photo"),
         ImageChooserPanel("article_photo"),
+        FieldPanel("table_of_contents"),
         StreamFieldPanel("body"),
     ]
 
+    @cached_property
+    def headers_list(self) -> List[str]:
+        list_of_headers = []
+        for stream_child in self.body:  # pylint: disable=not-an-iterable
+            if stream_child.block.name == "header":
+                list_of_headers.append(stream_child.value)
+        return list_of_headers
+
+    def get_header_id(self, title: str) -> int:
+        return self.headers_list.index(title) + 1
```

Comment:
Can you explain why you are adding here `1`?

Reply:
I wanted the IDs to start from 1 rather than 0. I don't feel like `id = 0` is something that makes sense 😕 , so I'm simply incrementing the index.

Reply:
But id 0 is something normal in computer science :D  That's why it made me curious why are you incrementing here it :P It's just convention that we're using here integers as ids. If we would use for example UUID it would might have then something like this: `aa19d760-243a-4061-9199-2b2b07e02ff5` and this is completely normal too :) Please delete it because it might confuse later too :)

Reply:
Ok. I just thought that while it's normal on the code side, it might be off-putting on the front-end side, which can be easily seen by a client.
But I'll remember not to worry about that 👍

Reply:
Done.